### PR TITLE
Support structured logging

### PR DESCRIPTION
**Pull request by saku3**
**What type of PR is this?**

 /kind feature

**What this PR does / why we need it**:

In this PR I have added JSON logging.
With this change, it is now possible to output the logs of the external-resizer container in JSON format.
Running the container with the --logging-format=json option will output the logs in JSON format.

In addition, I've modified the log messages based on the following guideline:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

I’ve update the klog functions in use according to the guidelines provided below, and I've confirmed that they pass the
[logcheck](https://github.com/kubernetes-sigs/logtools/tree/main/logcheck) tests:https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#change-log-functions

```
logcheck cmd/csi-resizer/main.go 
logcheck ./pkg/util/
logcheck ./pkg/resizer/ 
logcheck ./pkg/csi/    
logcheck ./pkg/controller/
logcheck ./pkg/features/ 
```

related PR:https://github.com/kubernetes-csi/livenessprobe/pull/202

**Special notes for your reviewer**:

```
# The logs below were outputted because the program was running on MacOS.
$ go run cmd/csi-resizer/main.go --logging-format=json
{"ts":1694049594780.344,"caller":"csi-resizer/main.go:105","msg":"Version","v":0,"version":"unknown"}
{"ts":1694049594780.369,"caller":"csi-resizer/main.go:128","msg":"Failed to create cluster config","err":"unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined"}
...

$ go run cmd/csi-resizer/main.go 
I0907 10:19:16.001980   70980 main.go:105] "Version" version="unknown"
E0907 10:19:16.002069   70980 main.go:128] "Failed to create cluster config" err="unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined"
...
```

**Does this PR introduce a user-facing change?**:

```release-note
Added support structured logging
```

### CODE REVIEW

File: cmd/csi-resizer/main.go [hunk 1]
```diff
@@ -112,15 +125,16 @@ func main() {
 		config, err = rest.InClusterConfig()
 	}
 	if err != nil {
-		klog.Fatal(err.Error())
+		klog.ErrorS(err, "Failed to create cluster config")
```
Comment (jsafrane) [inline cmd/csi-resizer/main.go:128]: Exit is missing here, `klog.Fatal` would exit but `ErrorS` does not.

Comment (saku3) [reply cmd/csi-resizer/main.go:128]: I've add `klog.FlushAndExit`. Could you please review it again?

**Comment by jsafrane** [comment]
/lgtm
@gnufied can you please take a look?

**Comment by saku3** [comment]
@jsafrane 

I've resolved the conflicts and pushed the changes according to the guidelines provided below. Could you please give it another look and approve if everything is fine?

https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync

**Comment by jsafrane** [comment]
/retest
It looks unrelated to this PR...

**Comment by saku3** [comment]
I'm getting the same error on multiple PRs in other repositories.
* external-snapshotter
* external-attacher
* external-provisioner

The error is identical.

```
{ failed [FAILED] Filesystem resize failed when restoring from snapshot to PVC with larger size. Restored fs size: 299853492224 bytes is not larger than Restored fs size: 299853492224 bytes is not larger than origin fs size: 299853492224 bytes.
HINT: Your driver needs to check the volume in NodeStageVolume and resize fs if needed.
HINT: For an example patch see: https://github.com/kubernetes/cloud-provider-openstack/pull/1563/files
Expected
    <int>: 299853492224
to be >
    <int>: 299853492224
In [It] at: test/e2e/storage/testsuites/provisioning.go:491 @ 10/02/23 18:08:27.298
}
```

**Comment by saku3** [comment]
Related to OSS issue: https://github.com/kubernetes/kubernetes/issues/120997

**Comment by saku3** [comment]
@jsafrane @gnufied 

Sorry to bother you again. I've resolved the conflicts and pushed the changes. Could you please give it another look and approve if everything is fine?

**Pull request merged**